spec.html

```diff
@@ -25895,8 +25925,54 @@ <h1>
         </emu-alg>
       </emu-clause>
 
+      <emu-clause id="sec-modulerequest-record">
+        <h1>ModuleRequest Records</h1>
+        <p>A <dfn id="modulerequest-record" variants="ModuleRequest Records">ModuleRequest Record</dfn> represents the request to import a module up to a given phase. It consists of the following fields:</p>
+        <emu-table id="table-modulerequest-record-fields" caption="ModuleRequest Record fields">
+          <table>
+            <tbody>
+              <tr>
+                <th>
+                  Field Name
+                </th>
+                <th>
+                  Value Type
+                </th>
+                <th>
+                  Meaning
+                </th>
+              </tr>
+              <tr>
+                <td>
+                  [[Specifier]]
+                </td>
+                <td>
+                  String
+                </td>
+                <td>
+                  The module specifier
+                </td>
+              </tr>
+              <tr>
+                <td>
+                  [[Phase]]
+                </td>
+                <td>
+                  ~source~ or ~evaluation~
+                </td>
+                <td>
+                  The target import phase
+                </td>
+              </tr>
+            </tbody>
+          </table>
+        </emu-table>
+
+        <emu-note type="editor">In general, this proposal replaces places where module specifiers are passed around with ModuleRequest Records. For example, several syntax-directed operations, such as ModuleRequests produce Lists of ModuleRequest Records rather than Lists of Strings which are interpreted as module specifiers. Some algorithms like ImportEntries and ImportEntriesForModule pass around ModuleRequest Records rather than Strings, in a way which doesn't require any particular textual change. Additionally, record fields in Cyclic Module Records and Source Text Module Records which contained Lists of Strings are replaced by Lists of ModuleRequest Records, as indicated above.</emu-note>
```

Comment:
The exact layout of this record depend on #3057. There is cleanup to be done around how and where exactly phases are passed around (and if they are part of this record), but this will be less merge conflict prone once Nicolo's PR is merged.

The PR is technically correct as is, so I'll defer this work until Import Attributes is landed.

Reply:
You'll also have to add `[[Phase]]` to LoadedModuleRequest Records.